app/models/container_label_tag_mapping.rb

```diff
@@ -12,5 +12,73 @@ class ContainerLabelTagMapping < ApplicationRecord
   #   We then also add a specific `label_value`->specific `tag` mapping here.
   belongs_to :tag
 
-  validates :label_name, :presence => true
+  # An empty string value is legal for labels; only NULL represents any value.
+  def any_value?
+    label_value.nil?
+  end
+
+  def specific_value?
+    !any_value?
+  end
+
+  def applies_to_label?(label)
+    ((labeled_resource_type.nil? || label.resource_type == labeled_resource_type) &&
+     label.name == label_name &&
+     (any_value? || label.value == label_value))
+  end
+
+  # Returns a Tag or nil.
+  def tag_for_label(label)
+    specific_mapping_for_value(label.value).tag if applies_to_label?(label)
+  end
+
+  # If this is an open ended any-value mapping, finds or creates a
+  # specific-value mapping to a specific tag.
+  def specific_mapping_for_value(value)
+    if specific_value?
+      self
+    else
+      self.class.where(
+        :labeled_resource_type => labeled_resource_type,
+        :label_name            => label_name,
+        :label_value           => value
+      ).first_or_create do |mapping|
+        mapping.tag = create_tag(value)
+      end
+    end
+  end
+
+  def create_tag(value)
```

Comment:
Maybe this was already discussed, I'm not sure dynamicly creating tags makes sense
I think tags are helpful in policies and other places if they are known in advance
Is it possible to do the tagging with the category tag? (cc @jrafanie)
you can always get the value from the custom attbibute